### PR TITLE
github cli v1.9.0

### PR DESCRIPTION
**Pull request by dscho**
This upgrades `gh` to v1.9.0, and while at it also allows running in WSL.

### CODE REVIEW

**Comment by dscho** [comment]
The UCRT64 build [fails thusly](https://github.com/msys2/MINGW-packages/pull/8376/checks?check_run_id=2351352654#step:8:66):

```
  ==> Missing dependencies:
    -> mingw-w64-ucrt-x86_64-go
  ==> ERROR: Could not resolve all dependencies.
```

I think fixing this problem is safely outside the purview of this PR ;-)

File: mingw-w64-github-cli/PKGBUILD [hunk 1]
```diff
@@ -16,29 +16,24 @@ checkdepends=("git")
 optdepends=("git: To interact with repositories")
 options=('!strip')
 source=("$_realname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
-        "0001-makefile-add-exe.patch"
```
Comment (naveen521kk) [inline mingw-w64-github-cli/PKGBUILD:19]: can you remove that patch from source tree? https://github.com/msys2/MINGW-packages/blob/062db83346b07d6f2dc5172a4311abadd2f3f1c9/mingw-w64-github-cli/gh

Comment (dscho) [reply mingw-w64-github-cli/PKGBUILD:19]: Whoops, yes, of course!

Comment (dscho) [reply mingw-w64-github-cli/PKGBUILD:19]: Done. Thank you for paying attention!

**Comment by naveen521kk** [comment]
looks like v1.9.1 is out now?

**Comment by jeremyd2019** [comment]
May get a v1.9.2 too... me and my 32-bit machine causing issues.  cli/cli#3431

**Comment by naveen521kk** [comment]
> May get a v1.9.2 too... me and my 32-bit os causing issues. cli/cli#3431

No worries, I will update when it is out :)

**Comment by dscho** [comment]
> 
> 
> looks like v1.9.1 is out now?

I have opened https://github.com/msys2/MINGW-packages/pull/8383, but...

> 
> 
> > May get a v1.9.2 too... me and my 32-bit os causing issues. [cli/cli#3431](https://github.com/cli/cli/issues/3431)
> 
> No worries, I will update when it is out :)

Indeed. I think we should wait a couple of days just in case v1.9.2 gets released. No need to overload the MSYS2 maintainers.

**Comment by lazka** [comment]
It's mostly automated, so PR away..

**Comment by dscho** [comment]
> 
> 
> It's mostly automated, so PR away..

Done.

**Pull request merged**